src/bun.js/node/node_fs.zig

```diff
@@ -2141,25 +2141,23 @@ pub const Arguments = struct {
 
             if (arguments.next()) |current| {
                 arguments.eat();
-                if (current.isNumber() or current.isBigInt()) {
-                    args.offset = current.to(u52);
-
-                    if (arguments.remaining.len < 2) {
+                if (current.isNumber() and arguments.remaining.len != 0) {
+                    var offset = current.to(i54);
+                    if (offset < 0) {
                         JSC.throwInvalidArguments(
-                            "length and position are required",
-                            .{},
+                            "The value of \"offset\" is out of range. It must be >= 0 && <= {}. Received {}",
+                            .{ std.math.maxInt(i54), offset },
                             ctx,
                             exception,
                         );
 
                         return null;
                     }
-                    if (arguments.remaining[0].isNumber() or arguments.remaining[0].isBigInt())
-                        args.length = arguments.remaining[0].to(u52);
 
-                    if (args.length == 0) {
+                    args.offset = current.to(u53);
+                    if (args.offset > args.buffer.buffer.len) {
                         JSC.throwInvalidArguments(
-                            "length must be greater than 0",
+                            "The value of \"offset\" should be lesser than buffer length",
```

Comment:
are you sure that node's behavior is to throw an error in this case and not clamp it?
```suggestion
                            "The value of \"offset\" should be less than the buffer length",
```

Reply:
Thank you for the review.

In node, for example:
If buffer length is 1000 and offset is 1100 and  length is 100 it would give error ```The value of "length" is out of range. It must be <= -100. Received 100.``` But it won't accept length less than 0.

Should I make this behave same as node?